captum/optim/_core/loss.py

```diff
@@ -8,21 +9,135 @@
 from captum.optim._utils.typing import ModuleOutputMapping
 
 
+def _make_arg_str(arg):
+    arg = str(arg)
+    too_big = len(arg) > 15 or "\n" in arg
+    return arg[:15] + "..." if too_big else arg
+
+
 class Loss(ABC):
     """
     Abstract Class to describe loss.
+    Note: All Loss classes should expose self.target for hooking by
+    InputOptimization
     """
 
-    def __init__(self, target: nn.Module) -> None:
+    def __init__(self) -> None:
         super(Loss, self).__init__()
-        self.target = target
+
+    @abstractproperty
+    def target(self):
+        pass
 
     @abstractmethod
-    def __call__(self, targets_to_values: ModuleOutputMapping):
+    def __call__(self, targets_to_values: ModuleOutputMapping) -> torch.Tensor:
         pass
 
+    def __repr__(self):
+        return self.__name__
+
+    def __add__(self, other):

```

Comment:
The *, /, **, +, -, and other operators come from the operator base class. So if we just call operators like that, we only need one function for the main module math stuff. Doing it this way should make it easy to allow any possible combination of math ops. 

Operator module link: https://docs.python.org/3/library/operator.html

Code example:

```
import operator


def module_op(self, other, math_op: Callable):
    if isinstance(other, (int, float)):

        def loss_fn(module):
            return math_op(other, self(module))

        name = self.__name__
        target = self.target
    else:
        # We take the mean of the output tensor to resolve shape mismatches
        def loss_fn(module):
            return math_op(torch.mean(self(module)), torch.mean(other(module)))

        name = f"Compose({', '.join([self.__name__, other.__name__])})"
        target = (
            self.target if hasattr(self.target, "__iter__") else [self.target]
        ) + (other.target if hasattr(other.target, "__iter__") else [other.target])
    return CompositeLoss(loss_fn, name=name, target=target)


class Loss(ABC):
    ...

    def __add__(self, other):
        return module_op(self, other, operator.add)

    def __mul__(self, other):
        return module_op(self, other, operator.mul)

    def __truediv__(self, other):
        return module_op(self, other, operator.truediv)

    def __pow__(self, other):
        return module_op(self, other, operator.pow)
```

I think that we also need to add `__rsub__`, and maybe other 'r' functions as well for full coverage.

Reply:
Using this now, thanks!